src/components/thumbnail/documents-type-collection.tsx

```diff
@@ -60,13 +60,18 @@ function getSectionDocs(section: NavTabSectionModelType, documents: DocumentsMod
           // shouldn't be published documents from other problems.
           const source = doc.originDoc || doc.uid;
           if (source) {
-            const entry = publishedDocs[source];
-            if (!entry || (entry.createdAt < doc.createdAt)) {
-              publishedDocs[source] = doc;
+            if (publishedDocs.source) {
+              publishedDocs.source.push(doc);
+            } else {
+              publishedDocs.source = [];
+              publishedDocs.source.push(doc);
             }
           }
         });
-        sectDocs.push(...Object.values(publishedDocs));
+      for (const sourceId in publishedDocs) {
+        const docsFromSource = publishedDocs[sourceId];
+        docsFromSource.forEach(d => sectDocs.push(d));
+      }
```

Comment:
I think you could also do something like:
```suggestion
      for (const sourceId in publishedDocs) {
        sectDocs.push(...publishedDocs[sourceId]);
      }
```